modules/tile-converter/src/i3s-converter/helpers/geometry-attributes.ts

```diff
@@ -195,35 +205,35 @@ function getSliceAttributeCount(
 function unifyObjectsByFeatureId(
   sortedData: GroupedByFeatureIdAttributes[]
 ): GroupedByFeatureIdAttributes[] {
-  const uniqueObjects: GroupedByFeatureIdAttributes[] = [];
-
-  for (let index = 0; index < sortedData.length; index++) {
-    const currentObject = sortedData[index];
-    const existingObject = uniqueObjects.find((obj) => obj.featureId === currentObject.featureId);
-
+  const groupedMetadata: {
+    featureId: number;
+    length: number;
+    attributes: GroupedByFeatureIdAttributes[];
+  }[] = [];
+  for (const data of sortedData) {
+    const existingObject = groupedMetadata.find((obj) => obj.featureId === data.featureId);
     if (existingObject) {
-      existingObject.positions = concatenateTypedArrays(
-        existingObject.positions,
-        currentObject.positions
-      );
-      existingObject.normals = concatenateTypedArrays(
-        existingObject.normals,
-        currentObject.normals
-      );
-      existingObject.colors = concatenateTypedArrays(existingObject.colors, currentObject.colors);
-      existingObject.texCoords = concatenateTypedArrays(
-        existingObject.texCoords,
-        currentObject.texCoords
-      );
-      existingObject.uvRegions = concatenateTypedArrays(
-        existingObject.uvRegions,
-        currentObject.uvRegions
-      );
+      existingObject.length += data.positions.length / 3;
```

Comment:
Maybe VALUES_PER_VERTEX  instead of 3 ?

Reply:
Removed `length` property. It is not needed anymore